operator/standalone/update.go

```diff
@@ -0,0 +1,386 @@
+		return err
+	}
+	updatePVCFunc := func() error {
+		u.persistentVolumeClaim.Spec.Resources.Requests = map[corev1.ResourceName]resource.Quantity{corev1.ResourceStorage: quantity}
+		return nil
+	}
+	_, err = controllerutil.CreateOrUpdate(ctx, u.client, u.persistentVolumeClaim, updatePVCFunc)
+	return err
+}
+
+func (u *UpdateStandalonePipeline) restartPostgresqlPod(ctx context.Context) error {
+	pod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      PostgresqlPodName,
+			Namespace: u.instance.Status.HelmChart.DeploymentNamespace,
+		},
+	}
+
+	// Restart the pod so that the new PVC storage takes effect
+	return u.client.Delete(ctx, pod)
+}
+
+func (u *UpdateStandalonePipeline) isStorageUpdated(_ context.Context) bool {
+	quantity, err := resource.ParseQuantity(u.instance.Spec.Parameters.Resources.StorageCapacity.String())
+	if err != nil {
+		return false
+	}
+	if *u.persistentVolumeClaim.Spec.Resources.Requests.Storage() != quantity {
+		return true
+	}
+	return false
+}
+
+func (u *UpdateStandalonePipeline) fetchPVC(ctx context.Context) error {
+	persistentVolumeClaim := &corev1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      getPVCName(),
+			Namespace: u.instance.Status.HelmChart.DeploymentNamespace,
+		},
+	}
+
+	err := getClientFromContext(ctx).Get(ctx, types.NamespacedName{Name: persistentVolumeClaim.Name, Namespace: persistentVolumeClaim.Namespace}, persistentVolumeClaim)
+	if err != nil {
+		return err
+	}
+	u.persistentVolumeClaim = persistentVolumeClaim
+	return nil
+}
+
+// isHelmReleaseReady returns true if the ModifiedTime is non-zero.
+//
+// Note: This only works for first-time deployments. In the future another mechanism might be better.
+// This step requires that fetchHelmRelease has run before.
+func (u *UpdateStandalonePipeline) isHelmReleaseReady(_ context.Context) bool {
+	if u.instance.Status.HelmChart != nil && !u.instance.Status.HelmChart.ModifiedTime.IsZero() {
+		return true
+	}
+	if u.helmRelease.Status.Synced {
+		if readyCondition := FindCrossplaneCondition(u.helmRelease.Status.Conditions, crossplanev1.TypeReady); readyCondition != nil && readyCondition.Status == corev1.ConditionTrue {
+			u.instance.Status.HelmChart.ModifiedTime = readyCondition.LastTransitionTime
+			return true
+		}
+	}
+	return false
+}
+
+// isPodRunning returns true if postgresql instance pod is running.
+//
+// Note: The pod may have been restarted during RunInitialUpdatePipeline thus
+// it is necessary to check if the pod is running
+func (u *UpdateStandalonePipeline) isPodRunning(ctx context.Context) bool {
+	pod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      PostgresqlPodName,
+			Namespace: u.instance.Status.HelmChart.DeploymentNamespace,
+		},
+	}
+	err := u.client.Get(ctx, types.NamespacedName{Name: pod.Name, Namespace: pod.Namespace}, pod)
+	if err == nil || isStatusConditionTrue(pod.Status.Conditions, conditions.TypeReady) {
+		return true
+	}
+	return false
+}
```

Comment:
With my suggestion to trigger a pod restart via annotation, we should check if the helm release exposes conditions that we can use to determine whether the instance is ready.

Reply:
We've tested this and adding the pod annotation is enough to trigger restarts (if the annotation changes, ofc)